creator-node/src/fileManager.js

```diff
@@ -748,6 +753,175 @@ async function removeFile(storagePath) {
   }
 }
 
+async function deleteFileOrDir(pathToFileOrDir) {
+  // Base case - delete single file (not a directory)
+  if (!(await fs.lstat(pathToFileOrDir)).isDirectory()) {
+    await fs.unlink(pathToFileOrDir)
+    return
+  }
+
+  // Recursively remove all contents of directory
+  for (const file of await fs.readdir(pathToFileOrDir)) {
+    const childPath = path.join(pathToFileOrDir, file)
+    if ((await fs.lstat(childPath)).isDirectory()) {
+      await deleteFileOrDir(childPath)
+    } else {
+      await fs.unlink(childPath)
+    }
+  }
+
+  // Remove actual directory
+  await fs.rmdir(pathToFileOrDir)
+}
+
+async function batchDeleteFileOrDir(storagePaths, batchSize, logger) {
+  let numFilesDeleted = 0
+  const batches = chunk(storagePaths, batchSize)
+  const promiseResults = []
+  for (const batchOfStoragePaths of batches) {
+    const promiseResultsForBatch = await Promise.allSettled(
+      batchOfStoragePaths.map((storagePath) => deleteFileOrDir(storagePath))
+    )
+    promiseResults.push(...promiseResultsForBatch)
+  }
+
+  // Count number of files successfully deleted
+  for (const promiseResult of promiseResults) {
+    if (promiseResult.status === 'fulfilled') {
+      numFilesDeleted++
+    } else {
+      logger.error(`Could not delete file: ${promiseResult?.reason?.stack}`)
+    }
+  }
+  return numFilesDeleted
+}
+
+/**
+ * Adds path to redis set for every file of the given user.
+ * DOES NOT DELETE. Call deleteAllCNodeUserDataFromDisk() to delete the data that was added to redis.
+ * Uses pagination to avoid loading all files in memory for users with a lot of data.
+ * @param {string} walletPublicKey the wallet of the user to delete all data for
+ * @param {bunyan.Logger} logger the logger to print messages to
+ * @return number of file paths added to redis
+ */
+async function gatherCNodeUserDataToDelete(walletPublicKey, logger) {
+  const FILES_PER_QUERY = 50_000
+  const redisSetKey = `${REDIS_DEL_FILE_KEY_PREFIX}${walletPublicKey}`
+  await redisClient.del(redisSetKey)
+
+  const cnodeUser = await models.CNodeUser.findOne({
+    where: { walletPublicKey }
+  })
+  if (!cnodeUser) throw new Error('No cnodeUser found')
+
+  const { cnodeUserUUID } = cnodeUser
+  logger.info(
+    `Fetching data to delete from disk for cnodeUserUUID: ${cnodeUserUUID}`
+  )
+
+  // Add files to delete to redis, paginated by storagePath, starting at the lowest real character (space)
+  let prevStoragePath = ' '
+  let numFilesAdded = 0
+  let files = []
+  do {
```

Comment:
could we use the code from here? https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/components/replicaSet/exportComponentService.js

Reply:
I'm not sure how I'd do that because those fetch within a clock range, whereas this query uses the previous storagePath for pagination and has these complicated nested selects. I think we'd have to use sequelize.literal, which is what [these docs](https://sequelize.org/docs/v6/other-topics/sub-queries/) do for nested queries anyway (plus we don't have any user-inputted data that we need to worry about sanitizing). wdyt?